src/main/scala/edu/ie3/mobsim/utils/PathsAndSources.scala

```diff
@@ -42,38 +41,123 @@ final case class PathsAndSources private (
     colSep: String
 )
 
+/** Beware: Since the simulation is executed as a jar within SIMONA relative
+  * paths are set according to your SIMONA project path. Furthermore if the
+  * output path is not configured it will be placed within the output simulation
+  * folder of the SIMONA run you are executing.
+  */
 object PathsAndSources extends LazyLogging {
 
-  /* Base path */
-  private val basePath: String = Paths.get("").toAbsolutePath.toString
-
-  private final case class InitData(
-      simulationName: String,
-      grid: InitData.Grid,
-      mobility: InitData.Mobility
-  )
-  private object InitData {
-    final case class Grid(name: String, path: String, colSep: String)
-    final case class Mobility(path: String, colSep: String)
-  }
+  private val basePath: Path = Paths.get("").toAbsolutePath
 
   def apply(
       simulationName: String,
-      inputConfig: MobSimConfig.Mobsim.Input
+      inputConfig: MobSimConfig.Mobsim.Input,
+      maybeOutputDir: Option[String]
   ): PathsAndSources = {
-    val initData = inputConfig match {
-      case Input(
-            Input.Grid(gridName, CsvParams(gridColSep, gridPath)),
-            Input.Mobility(CsvParams(mobColSep, mobPath))
-          ) =>
-        InitData(
-          simulationName,
-          InitData
-            .Grid(gridName, harmonizeFileSeparators(gridPath), gridColSep),
-          InitData.Mobility(harmonizeFileSeparators(mobPath), mobColSep)
+
+    val gridDir = harmonizeFileSeparators(inputConfig.grid.source.path)
+    val mobSimDir = harmonizeFileSeparators(inputConfig.mobility.source.path)
+    val gridColSep = inputConfig.grid.source.colSep
+
+    /* Build the different directory paths */
+    val mobSimDirPath = Paths.get(mobSimDir)
+    val mobSimInputDir =
+      if (mobSimDirPath.isAbsolute) mobSimDirPath.toString
+      else Seq(basePath, mobSimDir).mkString(File.separator)
+    val poiPath =
+      Seq(mobSimInputDir, "poi", inputConfig.grid.name, "poi.csv")
+        .mkString(File.separator)
```

Comment:
I think it could make sense to work with `File`s here, as you can extend them like this: `File file2 = new File(file1, "subdir");`. This would also add type safety (files are recognizable by type, as they're not just represented by Strings). I'd suggest opening a new issue for this

Reply:
#83